### ISSUE TITLE
Can I use aminal as an in-process terminal emulator?

### ISSUE THREAD

**Issue by firelizzard18** [liamg/darktile#281]
**Is your feature request related to a problem? Please describe.**
Similar to #118, is aminal embeddable? This is a separate issue, because I want to embed the terminal emulation part of aminal within a command-line tool. I do not want to use the windowed GUI (because I want my tool to be run within terminal/cmd).

**Describe the solution you'd like**
My goal is to have a cross-platform command-line tool that uses aminal for terminal emulation (like GNU screen, or tmux) with a custom transport layer.

Question 1, **can I use aminal with a custom transport layer?** Question 2, **can I embed aminal within a command-line tool?**

**Describe alternatives you've considered**
I've looked at using GNU screen, but A) it's not cross plaform and B) I don't know how to make it work with my custom transport, without all kinds of gymnastics.

**Additional context**
I am working on a specialized tool that more or less connects to a console on a remote device via a TCP/IP tunnel. I would like to use aminal as a front-end to this tunneled console. My transport protocol is very non-standard and I can guarantee that I won't be able to find existing tools for this.

**Comment by gcla** [comment]
Hi - I made a simple terminal widget as part of the gowid project, available at https://github.com/gcla/gowid. On the README is an example of a very simple terminal application that embeds this widget - a poor man's tmux. I'm not wholeheartedly recommending it, and if aminal was embeddable, I would absolutely try to connect it to gowid instead and replace what I did, but for now, it might be useful for you. My golang implementation is plagiarized from urwid's original python terminal widget - http://urwid.org/. It's VT220++, and vttest seems to render correctly. One user reported some problems with the display of wider runes like Chinese characters, and there's a branch called widerunes that fixes some of these issues. Looking forward to aminal's progress!
